types/json-rpc-ws/index.d.ts

```diff
@@ -0,0 +1,198 @@
+     *
+     * @param {String} address - url to connect to i.e. `ws://foo.com/`.
+     * @param {function} callback - optional callback to call once socket is connected
+     * @public
+     */
+    connect(url: string, connected: () => void): void;
+
+    /**
+     * Test whether we have a connection or not
+     *
+     * @returns {Boolean} whether or not we have a connection
+     * @public
+     */
+    isConnected(): boolean;
+
+    /**
+     * Return the current connection (there can be only one)
+     *
+     * @returns {Object} current connection
+     * @public
+     */
+    getConnection(): TConnection;
+
+    /**
+     * Close the current connection
+     *
+     * @param {function} callback - called when the connection has been closed
+     * @public
+     */
+    disconnect(callback: Function): void;
+}
+export declare class Base<TConnection extends Connection>
+{
+    /**
+     * Add a handler function for a given method
+     *
+     * @param {String} method - name of the method to add handler for.
+     * @param {function} handler - function to be passed params for given method.
+     * @todo enforce handler w/ two-param callback
+     * @public
+     */
+    expose<ParamType, ParamCallbackType>(eventName: string, handler: handler<TConnection, ParamType, ParamCallbackType>): void;
+    /**
+     * Send a method request through a specific connection
+     *
+     * @param {String} id - connection id to send the request through
+     * @param {String} method - name of method
+     * @param {Array} params - optional parameters for method
+     * @param {replyCallback} callback - optional reply handler
+     * @public
+     */
+    send<ParamType, ParamCallbackType>(eventName: string, params?: ParamType, callback?: replyCallback<ParamCallbackType>): void;
+    send<ParamType>(eventName: string, params: ParamType): void;
+    /**
+     * Connected event handler
+     *
+     * @param {Object} socket - new socket connection
+     * @private
+     */
+    connected(socket: ws): void;
+    /**
+     * Disconnected event handler
+     *
+     * @param {Object} connection - connection object that has been closed
+     * @private
+     */
+    disconnected(connection: TConnection): void;
+    /**
+     * Test if a handler exists for a given method
+     *
+     * @param {String} method - name of method
+     * @returns {Boolean} whether or not there are any handlers for the given method
+     * @public
+     */
+    hasHandler(method: string): boolean;
+
+    /**
+     * Get handler for a given method
+     *
+     * @param {String} method - name of method
+     * @returns {Array}  - handler for given method
+     * @public
+     */
+    getHandler<ParamType, ParamCallbackType>(method: string): handler<TConnection, ParamType, ParamCallbackType>;
+    /**
+     * Get a connection by id
+     *
+     * @param {id} id - id of the connection to get
+     * @returns {Connection} - Connection
+     * @public
```

Comment:
Don't include redundant JSDoc annotations -- accessibility and types are handled by TypeScript.
In this particular case I would say all of the docs are redundant.